### PR TITLE
fix or silence all lints

### PR DESCRIPTION
**Pull request by sigmaSd**
This is an attempt to get rid of the warnings

This PR:
- fix easy ones
- silence warnings for lines that seems can't be changed
- Add TODO for things I don't get yet

Overall is this a good thing ? I'm not really sure, but on the advantage at least now with 0 warnings you can add deno lint to ci, which would mandate it on PRs

### CODE REVIEW

File: mod.test.ts [hunk 1]
```diff
@@ -248,15 +248,16 @@ Deno.test("should handle boolean list 'and'", async () => {
 
 Deno.test("should support custom command handlers", async () => {
   const builder = new CommandBuilder()
+    // deno-lint-ignore require-await
     .registerCommand("zardoz-speaks", async (context) => {
```
Comment (dsherret) [inline mod.test.ts:252]: We could just remove the async keyword here I think? If not, we should make that possible to remove the ignore lint above.

File: mod.test.ts [hunk 2]
```diff
@@ -294,8 +295,9 @@ Deno.test("should support custom command handlers", async () => {
   }
 });
 
-Deno.test("should not allow invalid command names", async () => {
+Deno.test("should not allow invalid command names", () => {
   const builder = new CommandBuilder();
+  // deno-lint-ignore require-await
```
Comment (dsherret) [inline mod.test.ts:300]: Same here.

File: mod.test.ts [hunk 3]
```diff
@@ -635,10 +637,13 @@ Deno.test("basic logging test to ensure no errors", async () => {
   $.logDepth = 0;
 });
 
-Deno.test("setting logging", async () => {
+Deno.test("setting logging", () => {
   const test$ = $.build$();
+  // deno-lint-ignore no-explicit-any
```
Comment (dsherret) [inline mod.test.ts:642]: I really dislike this deno lint rule, but I get why people like it. Could we just add this to be ignored in deno.json?

File: src/shell.ts [hunk 4]
```diff
@@ -570,7 +582,7 @@ async function executeCommandArgs(commandArgs: string[], context: Context) {
   try {
     // ignore the result of writing to stdin because it may
     // have not finished before the process finished
-    const _ignore = writeStdin(context.stdin, p, completeSignal);
+    const _ignore = writeStdin(context.stdin, p, completeSignal); // TODO does this have to be assigned then ?
```
Comment (dsherret) [inline src/shell.ts:585]: It doesn''t have to be, but it's a floating promise. The `_ignore` is done to signify "yes, I know this is floating. Don't add an await here"

File: src/command.ts [hunk 5]
```diff
@@ -317,9 +321,10 @@ export class CommandBuilder implements PromiseLike<CommandResult> {
         case "null":
         case "piped":
           return kind;
-        default:
-          const _assertNever: never = kind;
+        default: {
+          const _assertNever: never = kind; //TODO whats this trick
```
Comment (dsherret) [inline src/command.ts:325]: It's a way to assert that `kind` is a `never` type. If you remove one of the cases above or add another value to `ShellPipeWriterKind` then it should error.

File: src/commands/export.ts [hunk 6]
```diff
@@ -1,10 +1,12 @@
 import { CommandContext } from "../command_handler.ts";
 import { EnvChange, ExecuteResult } from "../result.ts";
 
+// All commands needs to have async defintion
+// deno-lint-ignore require-await
```
Comment (dsherret) [inline src/commands/export.ts:5]: Again, we could remove this and allow synchronous commands.

File: src/common.ts [hunk 7]
```diff
@@ -77,6 +77,7 @@ export class TreeBox<T> {
   }
 
   getValue() {
+    // deno-lint-ignore no-this-alias
```
Comment (dsherret) [inline src/common.ts:80]: Seems annoying. Let's ignore this in the deno.json file too.

File: mod.test.ts [hunk 8]
```diff
@@ -248,15 +248,16 @@ Deno.test("should handle boolean list 'and'", async () => {
 
 Deno.test("should support custom command handlers", async () => {
   const builder = new CommandBuilder()
+    // deno-lint-ignore require-await
     .registerCommand("zardoz-speaks", async (context) => {
       if (context.args.length != 1) {
-        await context.stderr.writeLine("zardoz-speaks: expected 1 argument");
+        context.stderr.writeLine("zardoz-speaks: expected 1 argument");
```
Comment (dsherret) [inline mod.test.ts:254]: Thanks! This used to be async.

**Review by dsherret — COMMENTED** [review]
Thanks @sigmaSd! Just a few comments.

File: src/commands/export.ts (hunk 6 above)
Comment (sigmaSd) [reply src/commands/export.ts:5]: So all commands should be comes sync ? or is there a way to make it both sync and async ?
Current signature https://github.com/dsherret/dax/blob/main/src/command_handler.ts#L23

**Comment by sigmaSd** [comment]
Ah so the formatting keeps state or something, I'm used to rust fmt, where it doesn't matter the code state the formatting will always be the same, so I actually need to revert formatting manually then xD

File: src/commands/export.ts (hunk 6 above)
Comment (dsherret) [reply src/commands/export.ts]: Actually you could just use Promise.resolve (can’t respond much atm)

Comment (sigmaSd) [reply src/commands/export.ts]: Don't worry about it, when you have time feel free to review it

I went with making commandhandler ` Promise<ExecuteResult> |  ExecuteResult` and when it gets called I just await (I can branch on its type promise or not)

**Comment by sigmaSd** [comment]
I rebased the code, made command handler async | sync, the result is being awaited in either case https://github.com/dsherret/dax/blob/main/src/shell.ts#L546

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**